nixos/modules/services/misc/gitea.nix

```diff
@@ -454,6 +471,41 @@ in
       ];
     };
 
+    services.nginx = lib.mkIf cfg.configureNginx {
+      enable = true;
+      virtualHosts."${cfg.domain}" = {
+        root = "/var/empty";
+
+        locations."/" = {
+          proxyPass = (if cfg.enableUnixSocket then "http://unix:/run/gitea/gitea.sock" else "http://127.0.0.1:${toString(cfg.httpPort)}");
+          recommendedProxySettings = true;
+        };
+
+        locations."/avatars/" = {
+          alias = "${cfg.stateDir}/data/avatars/";
+          tryFiles = "$uri =404";
+
+          extraConfig = ''
+            default_type "image";
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."${cfg.settings.server.STATIC_URL_PREFIX}/assets/" = {
+          alias = "${cfg.package.data}/public/";
+          tryFiles = "$uri =404";
```

Comment:
I'm not sure if I understand why these locations are actually needed here.

Reply:
Allows to push files directly through nginx without redirecting the request to gitea.

Reply:
Just to clarify: Those are the paths as outlined in https://docs.gitea.io/en-us/reverse-proxies/#nginx-and-serve-static-resources-directly (implemented 2019)

Gitea can serve static files just as well, obviously (as most users do).

Funnily enough, neither the official https://gitea.com nor https://codeberg.org seem to use that feature.
No idea if it actually matters all that much /shrug